src/test/java/io/confluent/connect/jdbc/sink/integration/JdbcSinkConnectorIT.java

```diff
@@ -0,0 +1,277 @@
+    count = loadFromSQL(KAFKA_TOPIC);
+    Assert.assertEquals(totalRecords.count(), NUM_RECORDS * 2);
+    Assert.assertEquals(NUM_RECORDS * 2, count);
+  }
+
+  @Test
+  public void testWithJDBCMysqlDialectUnavailable() throws Exception {
+
+    dropTableIfExists(KAFKA_TOPIC);
+    sendTestDataToKafka(SCHEMA, 0);
+
+    // Add mysql dialect related configurations.
+    props.put("connection.url", "jdbc:mysql://localhost:3306/db");
+    props.put("connection.user", "user");
+    props.put("connection.password", "password");
+    props.put("dialect.name", "MySqlDatabaseDialect");
+    props.put("auto.create", "true");
+    props.put("value.converter", JsonConverter.class.getName());
+
+    // Start Connector and wait some specific time to start the connector.
+    connect.configureConnector(CONNECTOR_NAME, props);
+    waitForConnectorToStart(CONNECTOR_NAME, Integer.valueOf(MAX_TASKS));
+    Thread.sleep(3000);
+    ConsumerRecords<byte[], byte[]> totalRecords = connect.kafka().consume(
+        NUM_RECORDS,
+        CONSUME_MAX_DURATION_MS,
+        KAFKA_TOPIC);
+
+    log.info("Number of records added in kafka {}", totalRecords.count());
+    int count = loadFromSQL(KAFKA_TOPIC);
+    Assert.assertEquals(NUM_RECORDS, count);
+    sendTestDataToKafka(SCHEMA, 1);
+    totalRecords = connect.kafka().consume(
+        NUM_RECORDS * 2,
+        CONSUME_MAX_DURATION_MS,
+        KAFKA_TOPIC);
+    count = loadFromSQL(KAFKA_TOPIC);
+    Assert.assertEquals(totalRecords.count(), NUM_RECORDS * 2);
+    Assert.assertEquals(NUM_RECORDS * 2, count);
+  }
+
+  @Test
+  public void testWithChangeInServiceCredentials() throws Exception {
+    grantAllPrivileges();
+    dropTableIfExists(KAFKA_TOPIC);
+    sendTestDataToKafka(SCHEMA, 0);
+    // Add mysql dialect related configurations.
+    props.put("connection.url", "jdbc:mysql://localhost:3306/db");
+    props.put("connection.user", "user");
+    props.put("connection.password", "password");
+    props.put("dialect.name", "MySqlDatabaseDialect");
+    props.put("auto.create", "true");
+    props.put("value.converter", JsonConverter.class.getName());
+
+    // Start Connector and wait some specific time to start the connector.
+    connect.configureConnector(CONNECTOR_NAME, props);
+    waitForConnectorToStart(CONNECTOR_NAME, Integer.valueOf(MAX_TASKS));
+    ConsumerRecords<byte[], byte[]> totalRecords = connect.kafka().consume(
+        NUM_RECORDS,
+        CONSUME_MAX_DURATION_MS,
+        KAFKA_TOPIC);
+
+    log.info("Number of records added in kafka {}", totalRecords.count());
+    int count = loadFromSQL(KAFKA_TOPIC);
+    Assert.assertEquals(NUM_RECORDS, count);
+    revokeWritePrivileges();
+    Thread.sleep(10000);
+    sendTestDataToKafka(SCHEMA, 1);
+    totalRecords = connect.kafka().consume(
+        NUM_RECORDS * 2,
+        CONSUME_MAX_DURATION_MS,
+        KAFKA_TOPIC);
+    count = loadFromSQL(KAFKA_TOPIC);
+    Assert.assertEquals(totalRecords.count(), NUM_RECORDS * 2);
```

Comment:
If we are revoking the write privilages how are we receiving the records in mysql?